### PR TITLE
Add SonarCloud analysis action

### PR DESCRIPTION
**Pull request by fmarco76**
Added the action to perform SonarCloud code analysis. In order to work properly the SonarCloud token for the project must be added among the secrets.

Additionally, the build steps are isolated in callable work-flow. It seems the steps are almost identical for the other work-flows so they could be modified to call the same build steps and avoid to repeat the same steps everywhere.

### CODE REVIEW

**Comment by fmarco76** [comment]
Coverage is not addressed because it has to be integrated with the build system and there are not guideline for cmake

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Sonarcloud
+on: [push,pull_request]
+
+jobs:
+  call-build-workflow:
+    uses: ./.github/workflows/build.yml
+    secrets: inherit
```
Comment (ckelleyRH) [inline .github/workflows/sonarcloud.yml:7]: Does this allow you to call a workflow form inside a workflow? If so this is a game changing discovery!

**Review by ckelleyRH — APPROVED** [review]
LGTM!

I am able to see the result of the run against this PR [here](https://sonarcloud.io/organizations/dogtagpki/projects) - let me know if you can't access it.

**Comment by ckelleyRH** [comment]
> Coverage is not addressed because it has to be integrated with the build system and there are not guideline for cmake

Makes sense. FYI I have set the quality gating to mirror what exists for JSS already - on the next run the gating should happen (and it will fail).

**Comment by ckelleyRH** [comment]
It's weird that this job fails though: https://github.com/dogtagpki/pki/runs/6757890990?check_suite_focus=true

**Comment by fmarco76** [comment]
It fails because the SonarCloud token is missed. I have not the right to add secret to the repository. You have to create the secret. If you go to SonarCloud and open the configuration for pki project you can get the token to configure here

**Review by edewata — APPROVED** [review]
I have some comments, but I'm not too familiar with SonarCloud so feel free to merge if you think it's fine.

**Comment by edewata** [comment]
Just FYI, right now in each test workflow we only build the packages required for that workflow, but I think we can just build all packages in all cases so we can reuse this build workflow. I don't think the extra time required to build all packages will be significant. Feel free to address this in a separate PR.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          # keep packages after installation
+          echo "keepcache=True" >> /etc/dnf/dnf.conf
+          dnf install -y dnf-plugins-core rpm-build moby-engine
+          dnf copr enable -y ${{ needs.init.outputs.repo }}
+          dnf builddep -y --allowerasing --spec ./pki.spec --nogpgcheck
+          # don't cache COPR packages
+          rm -f `find /var/cache/dnf -name '*.rpm' | grep '/var/cache/dnf/copr:'`
+
+      - name: Build PKI packages
+        run: ./build.sh --with-pkgs=base,server,ca,tests --with-timestamp --work-dir=build rpm
```
[the remark this answers is not in the document]
Comment (fmarco76) [reply .github/workflows/build.yml]: OK

**Pull request merged**